index.bs

```diff
@@ -5638,6 +5639,97 @@ At this time, one [=credential property=] is defined: the [=resident key credent
 : Authenticator extension output
 :: None.
 
+## Pseudo-random function extension (<dfn>prf</dfn>) ## {#prf-extension}
+
+This [=client extension|client=] [=registration extension=] and [=authentication extension=] allows a [=[RP]=] to evaluate outputs from a pseudo-random function (PRF) associated with a [=credential=]. The PRFs provided by this extension map from {{USVString}}s of any length to 32-byte {{ArrayBuffer}}s.
+
+When requested, if a [=credential=] is created on an [=authenticator=] that supports this extension then either one or two freshly-seeded PRFs are associated with it. If two PRFs are associated then, when outputs are requested, one PRF is evaluated if [=user verification=] is performed and the other is evaluated if only a [=test of user presence|user presence test=] is performed. It is the responsibility of the [=[RP]=] to set the {{PublicKeyCredentialRequestOptions/userVerification}} parameter accordingly: [=[RPS]=] SHOULD *consistently* use either the {{UserVerificationRequirement/required}} or {{UserVerificationRequirement/discouraged}} values of {{UserVerificationRequirement}} when using this extension, otherwise the outputs may vary for a given input. To avoid mistakes, the default value of {{UserVerificationRequirement/preferred}} is prohibited when using this extension during [=assertion=] because that could cause the PRF used to vary between operations if [=user verification=] is later enabled on an [=authenticator=].
```

Comment:
It's not clear if the "first" or "second" prf is the one returned during this exension. This must be clarified to state which of "first" or "second" are returned, and in which userverification scenarios related to the registration process.

Reply:
(better yet, rename these from "first" and "second" to "default" and "verifiedonly"

Reply:
It seems like you might be confusing the "two PRFs" with the two `first` and `second` extension output fields. Is that so? The `first` and `second` outputs are both from the same PRF, and which PRF is determined by whether UV was used:

```
f = if UV=1 { PRF1 } else { PRF2 }
output.first = f(input.first)
if input.second {
  output.second = f(input.second)
}
```

I thought this was unambiguously defined by the introduction:

>[...] If two PRFs are associated then, when outputs are requested, one PRF is evaluated if _user verification_ is performed and the other is evaluated if only a _user presence test_ is performed. [...]

and the definition of the `results` output:

>The results of evaluating the PRF for the inputs given in `eval` or `evalByCredential`. Outputs may not be available during registration; see comments in `eval`.

How would you suggest clarifying it? Would a clarification also resolve your [previous comment](https://github.com/w3c/webauthn/pull/1424#discussion_r617957792)?